### PR TITLE
Add lookupSourcePositions to JoinOperatorInfo

### PR DESCRIPTION
**Pull request by raghavsethi**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSource.java [hunk 1]
```diff
@@ -257,7 +257,7 @@ public Factory(List<Supplier<LookupSource>> partitions)
                         .toArray(LookupSource[]::new);
 
                 visitedPositions = Arrays.stream(this.lookupSources)
-                        .map(LookupSource::getJoinPositionCount)
+                        .map(s -> toIntExact(s.getJoinPositionCount()))
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSource.java]: nit: you could add `.map(Math::toIntExact)` instead :)

File: presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSource.java [hunk 2]
```diff
@@ -105,9 +105,9 @@ public int getChannelCount()
     }
 
     @Override
-    public int getJoinPositionCount()
+    public long getJoinPositionCount()
     {
-        throw new UnsupportedOperationException("Parallel hash can not be used in a RIGHT or FULL outer join");
+        return Arrays.stream(lookupSources).mapToLong(LookupSource::getJoinPositionCount).sum();
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSource.java]: nit: i prefer `.map` and `.sum` on separate lines (but can remain as is)

File: presto-main/src/main/java/com/facebook/presto/operator/JoinOperatorInfo.java [hunk 3]
```diff
@@ -29,29 +29,32 @@
...
+    public static JoinOperatorInfo createJoinOperatorInfo(JoinType joinType, long[] logHistogramCounters, long lookupSideRows)
     {
         long[] logHistogramProbes = new long[HISTOGRAM_BUCKETS];
         long[] logHistogramOutput = new long[HISTOGRAM_BUCKETS];
         for (int i = 0; i < HISTOGRAM_BUCKETS; i++) {
             logHistogramProbes[i] = logHistogramCounters[2 * i];
             logHistogramOutput[i] = logHistogramCounters[2 * i + 1];
         }
-        return new JoinOperatorInfo(joinType, logHistogramProbes, logHistogramOutput);
+        return new JoinOperatorInfo(joinType, logHistogramProbes, logHistogramOutput, lookupSideRows);
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/JoinOperatorInfo.java]: I'd name this `lookupSourcePositions` to be consistent with the other constructors.

File: presto-main/src/main/java/com/facebook/presto/operator/JoinStatisticsCounter.java [hunk 4]
```diff
@@ -34,12 +34,18 @@
     //      [2*bucket]      count probe positions that produced "bucket" rows on source side,
     //      [2*bucket + 1]  total count of rows that were produces by probe rows in this bucket.
     private final long[] logHistogramCounters = new long[HISTOGRAM_BUCKETS * 2];
+    private long lookupSourcePositions = 0;
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/JoinStatisticsCounter.java]: `0` is the default value

**Pull request closed without merging**